packages/@react-aria/searchfield/docs/useSearchAutocomplete.mdx

```diff
@@ -370,17 +368,25 @@ function Example() {
     {id: 8, name: 'Agricultural'},
     {id: 9, name: 'Electrical'}
   ];
-  let [majorId, setMajorId] = React.useState();
+  let [major, setMajor] = React.useState();
+
+  let onSubmit = (value, key) => {
+    if (value) {
+      setMajor(value);
+    } else if (key) {
+      setMajor(options.find(o => o.id === key).name);
+    }
+  };
 
   return (
     <>
       <SearchAutocomplete
         label="Search engineering majors"
         defaultItems={options}
-        onSelectionChange={setMajorId}>
+        onSubmit={onSubmit}>
         {(item) => <Item>{item.name}</Item>}
```

Comment:
Think you need pass the key to the Item so that your onSubmit can dig up the right major to set. Right now nothing is set when you hit enter on a major in the list

Reply:
Added to follow-up (#2286)